internal/controller/controllers/infraenv_controller.go

```diff
@@ -366,23 +374,27 @@ func (r *InfraEnvReconciler) ensureISO(ctx context.Context, log logrus.FieldLogg
 	}
 	infraEnvInternal, err := r.Installer.GetInfraEnvByKubeKey(key)
 	if err != nil {
+		// If the infraenv does not exist then create it.
 		if errors.Is(err, gorm.ErrRecordNotFound) {
 			var clusterID *strfmt.UUID
 			var openshiftVersion string
 			if cluster != nil {
 				clusterID = cluster.ID
 				openshiftVersion = cluster.OpenshiftVersion
 			}
-			infraEnvInternal, err = r.createInfraEnv(ctx, log, &key, infraEnv, clusterID, openshiftVersion)
+			// The OpenShift installer validation code for the additional trust bundle
+			// is buggy and doesn't react well to additional newlines at the end of the
+			// certs. We need to strip them out to not bother assisted users with this
+			// quirk.
+			additionalTrustBundle := strings.TrimSpace(infraEnv.Spec.AdditionalTrustBundle)
```

Comment:
I don't think this should be done here. Can we do this when we get the internal request to create the infraenv instead of in the controller?